### ISSUE TITLE
independent.co.uk

### ISSUE THREAD

**Issue by gotitbro** [jspenguin2017/uBlockProtector#616]
**uBlock Protector** currently tries to bypass The Independent's anti adb measure by making it believe that ads were loaded ([Solution](https://github.com/jspenguin2017/uBlockProtector/blob/c8df534bcb5f7a8e68e73b3ac2da788df38443fb/Extension%20Compiler/Extension/content/3-content-rules-2-specific.js#L601)).

This leads to an issue where The Independent's website displays the "**_Thank you for supporting independent.co.uk_**" banner each and every time the website is visited making it really inconvenient for users.

I recommend removing the fix for The Independent. I don't even see the anti adb message anymore on the website if you do it would be better just to hide it.

**SS:** https://i.imgur.com/IpZtFeP.png (Thank You banner shown every time website is visited)

**Error URL:** http://www.independent.co.uk/life-style/gadgets-and-tech/google-birthday-surprise-spinner-date-problem-start-company-a7968951.html

_**Note:** Using uBO 1.14.8 on Chrome 61 with the default filters and settings along with uBlock Protector._

**Comment by jspenguin2017** [comment]
I can't reproduce it, and the rule you kinked would match `puzzles.independent.co.uk` but not `www.independent.co.uk`. 
Can you try clearing cache and cookies of the website?

**Comment by uBlock-user** [comment]
Can't reproduce.

**Comment by uBlock-user** [comment]
Rules on sub-domains are never applied on the main domain, so that's not the cause of this.

**Comment by gotitbro** [comment]
@jspenguin2017 Sorry linked the wrong blob, fixed now. 

You mean you don't see the Thank You notice at all? I see it every time and that is definitely due to uBlock Protector.

**Comment by jspenguin2017** [comment]
I don't see the thank you notice. 
![image](https://user-images.githubusercontent.com/7283682/30915505-3db33b5c-a354-11e7-8b90-7a4a52ab1497.png)

**Comment by jspenguin2017** [comment]
```js
(function($) {
    $('body').on('AdblockDetected', function() {
        // check for mobile with helper
        if (typeof window.helper.isMobile !== 'undefined' && !window.helper.isMobile.any()) {
            var adblock_cookie = window.helper.getCookie('adblock_detected');
            if (adblock_cookie !== 'ignored') {
                $('.adblock.adblock-modal').css('display', 'block').appendTo('body');
                window.helper.createCookie('adblock_detected', 'detected', 1);
                $('.adblock-modal [data-action="close"]').click(function(e) {
                    window.helper.createCookie('adblock_detected', 'ignored', 30);
                    $('.adblock-modal').remove();
                });
                $(window).resize(function() {
                    var height = $(window).height();
                    var container = $('.adblock-modal .adblock-container').filter(':visible');
                    container.height('auto');
                    container.css('overflow-y', '');
                    if (container.height() > $(window).height() - 190) {
                        container.height($(window).height() - 190);
                        container.css('overflow-y', 'scroll');
                    }
                });
                $(window).trigger('resize');
            }
        }
    });
})(jQuery);
```
uBlock Protector will make the website thinks that you clicked the close button which dismissed the anti-adblock modal, that isn't the cause. You probably have some conflicting extensions.

**Comment by gotitbro** [comment]
@jspenguin2017 I don't have any extensions enables besides **_uBO, uBO Extra & uBlock Protector_**. I made a video showing the problem: https://streamable.com/wnyw9

**Comment by jspenguin2017** [comment]
You are not on standard setup aren't you? The modal should be blocked by `Adblock Warning Removal List`. But you are right that the script rule is redundant.

**Comment by uBlock-user** [comment]
Remove uBO-Extra, it's already covered with uBlock Protector.

**Comment by gotitbro** [comment]
@jspenguin2017 I was not aware there was a "standard setup". What exactly that might be?

@uBlock-user uBO Extra does not cause any issues and it lets me see websocket requests in logger and efficiently blocks Instart Logic.

**Comment by jspenguin2017** [comment]
Here is the standard setup: 
https://github.com/jspenguin2017/uBlockProtector/blob/master/Notes/Standard%20Configuration.MD

uBO-Extra is embedded already, although running another copy shouldn't cause problem (anymore). 
WebSocket part is disabled on modern Chromium. 
https://github.com/jspenguin2017/uBlockProtector/blob/master/Extension%20Compiler/Extension/content/5-ubo-extra.js

**Comment by uBlock-user** [comment]
@gotitbro So does uBlock Protector as uBO-Extra is embedded inside the extension.

> it lets me see websocket requests in logger

You don't need uBO-Extra anymore as with v58, websocket connections are now natively passed by the browser to the webRequest API, read - https://github.com/gorhill/uBO-Extra/wiki/Sites-on-which-uBO-Extra-is-useful#websocket-connection-attempts

**Comment by gotitbro** [comment]
@jspenguin2017 @uBlock-user Thanks for the info. I didn't know that uBO Extra was already embedded inside uBlock Protector. Why was this decision made? 

I would still like to have it as a separate extension though for faster updates and resolutions.

**Comment by uBlock-user** [comment]
Your choice though, but if it ever was to cause any conflict on any website you're visiting, this would be the first potential cause.

This is the std setup -  https://github.com/jspenguin2017/uBlockProtector/blob/master/Notes/Standard%20Configuration.MD#the-standard-configuration

**Comment by gotitbro** [comment]
@uBlock-user Well it shouldn't according the above post by @jspenguin2017.

**Comment by uBlock-user** [comment]
On this website, yes, I meant it in general. There's no guarantee that it won't cause any conflict on any websites.

**Comment by gotitbro** [comment]
I hope it doesn't, I still would like to use it for the time being as for the reason mentioned above.

**Comment by jspenguin2017** [comment]
You can use uBO-Extra, just make sure to install it before uBlock Protector so its rule can take effect. It shouldn't cause conflict, not anymore. You might see more error logs in the console but it's just because a rule can't be applied twice, the website will work as expected. 
I do pull update from uBO-Extra repository every day, you would get faster update by installing uBO-Extra itself, but not by much. 
I have uBO-Extra embedded because it gives me a bit more control this way, in case it causes problems, I can add a white list until it is addressed on uBO-Extra side.

**Comment by uBlock-user** [comment]
Speaking of Instart Logic, I found blowing up `atob` works fine too so you're not totally dependent on uBO-Extra. Gorhill locked up the topic so I couldn't post anymore, but here's the filter if you're interested in an alternate solution for IL. Also `abort-on-property-read/write` are not affected by race conditions either FYI

`sandiegouniontribune.com##script:inject(abort-on-property-read.js, atob)`

**Comment by jspenguin2017** [comment]
I thought we already concluded that the race condition makes any `script:inject` rules ineffective against IL?

**Comment by uBlock-user** [comment]
From what I understand, it doesn't check for specific condition to be met for it to block a specific inline script and declares the property atob to be restricted from being accessed that's it. So it always win the race instead. If I disable the Extension and add that filter, `g00` is never inserted on my end atleast.

**Comment by jspenguin2017** [comment]
`abort-current-inline-script.js` declares a hook as well, that's not the problem. As far as I know, there is no way for uBO to synchronously inject scriptlet, not on Chrome, that's why uBO-Extra exists, and that's why uBlock Protector is not configurable.

**Comment by gotitbro** [comment]
@jspenguin2017 So even if I remove uBO Extra I shouldn't see any ads that uBO Extra currently removes or will remove right? There would be no difference in performance? 

And any issues that get resolved in uBO Extra shouldn't take more than a day to be integrated into uBlock Protector?

**Comment by gotitbro** [comment]
@jspenguin2017 I also noticed that "**_uBlock Protector list_**" was removed from uBO and replaced with "**_AAK-Cont Filters For uBlock Origin_**". Which one should we be using and why was this change initiated?

**Comment by jspenguin2017** [comment]
The performance gain is minimal, but yes, the solutions are synced daily. 
You should still use uBlock Protector List. The change was because uBlock Protector only works on Chrome.

**Comment by uBlock-user** [comment]
In gorhill's own words "uBO should be browser vendor agnostic", that's why he removed and replaced it with AAK-Cont.

**Comment by gotitbro** [comment]
@jspenguin2017 So there shouldn't be any noticeable difference in usage for me even if I solely use uBlock Protector right? And what is the difference between AAK-Cont and uBlock Protector list anyway?

**Comment by jspenguin2017** [comment]
There will be noticeable difference, AAK-Cont is a lot less reliable.

**Comment by gotitbro** [comment]
> There will be noticeable difference, AAK-Cont is a lot less reliable.
Sorry I should've been clearer. I meant  would there be any noticeable difference if I uninstall uBO Extra?

**Comment by jspenguin2017** [comment]
No, the difference is too small to be noticeable.

**Comment by gotitbro** [comment]
@jspenguin2017 Then I am just going to remove uBO Extra as I see that no major benefit will be provided by keeping it. Hope everything works just as well as it does now.

**Comment by ghajini** [comment]
@uBlock-user,
I see your filter about IL ***sometimes*** work
https://prnt.sgrgksc

https://prnt.sc/grgksx

This i10c referrer is always stick to url in url bar......but i don't see any ads /g00 cookies....
then I came further browse site with your filter for almost 30 mins like opening pages in new tab/current tab n opening page stories....I got this...
Instance of g00 referrer is almost 5 times as seen from ubo logger.....
https://prnt.sc/grgktn 
https://prnt.sc/grgkt5 
Then almost after little more clicking stories to read, it send me to ***classic*** g00 referring cookies (about 150-200 )stored in browser......CPU memory consumption increased and so on.....

So your filter needs little improvisation in order to counter IL....

As contrast , ubo extra is working very fast whether you click on any story ,it never redirects you to g00 /IL,  no matter how much time you browse....it always stays at top of page and prevents IL from executing at all......

**Comment by jspenguin2017** [comment]
@ghajini Filters cannot overcome uBO limitations. Need to wait for Chromium to implement response body replacing API.
